cmake/nuttx/px4_impl_nuttx.cmake

```diff
@@ -425,6 +425,8 @@ function(px4_nuttx_add_romfs)
 	endforeach()
 	add_custom_target(collect_extras DEPENDS ${extras})
 
+	string(REPLACE ";" "\ " gcc_c_flags_2 "${gcc_c_flags}")
```

Comment:
Maybe I don't quite understand what's happening here, but could you move most of this to Toolchain-arm-none-eabi-clang.cmake and then these flags would already be set in CMAKE_C_FLAGS?